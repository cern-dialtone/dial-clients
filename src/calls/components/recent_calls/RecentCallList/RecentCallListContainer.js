import { connect } from 'react-redux';
import RecentCallList from './RecentCallList';

function mapStateToProps({ recent }) {
  return {
    recentCalls: recent.recentCalls
  };
}

export default connect(mapStateToProps)(RecentCallList);
