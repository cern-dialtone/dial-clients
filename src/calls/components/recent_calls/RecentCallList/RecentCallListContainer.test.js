import React from 'react';
import { render, waitForElement } from '@testing-library/react';

import { Provider } from 'react-redux';

import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';

import { createHashHistory } from 'history';
import { routerMiddleware } from 'connected-react-router';

import RecentCallListContainer from './RecentCallListContainer';

const history = createHashHistory();
const middlewares = [thunk, routerMiddleware(history)];
const mockStore = configureMockStore(middlewares);

describe('RecentCallListContainer test', () => {
  let store;
  let initialState;

  beforeEach(() => {
    initialState = {
      recent: {
        recentCalls: []
      }
    };
    store = mockStore(initialState);
    store.dispatch = jest.fn();
  });

  it('renders empty list', async () => {
    const { getByTestId, asFragment } = render(
      <Provider store={store}>
        <RecentCallListContainer t={key => key} />
      </Provider>
    );
    await waitForElement(() => getByTestId('RecentCallList'));
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders a List without any errors', async () => {
    initialState.recent.recentCalls = [
      {
        name: '12345',
        endTime: 12345,
        phoneNumber: '222333',
        missed: true,
        incoming: true
      },
      {
        name: '123456',
        endTime: 12345,
        phoneNumber: '444555',
        missed: false,
        incoming: false
      }
    ];

    const { getByTestId, asFragment } = render(
      <Provider store={store}>
        <RecentCallListContainer t={key => key} />
      </Provider>
    );
    await waitForElement(() => getByTestId('RecentCallList'));
    await waitForElement(() => getByTestId('recent-0'));
    await waitForElement(() => getByTestId('recent-1'));
    expect(asFragment()).toMatchSnapshot();
  });
});
