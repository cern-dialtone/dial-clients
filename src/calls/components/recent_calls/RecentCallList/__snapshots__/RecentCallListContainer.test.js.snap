// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecentCallListContainer test renders a List without any errors 1`] = `
<DocumentFragment>
  <div
    class="ScrollableContent"
  >
    <div
      class="ui link items"
      data-testid="RecentCallList"
    >
      <div
        class="item padded-item RecentCall"
        data-testid="recent-0"
      >
        <i
          aria-hidden="true"
          class="grey phone large icon ui avatar"
        />
        <div
          class="content"
        >
          <div
            class="header ItemHeader"
          >
            12345
          </div>
          <div
            class="description Content"
          >
            <i
              aria-hidden="true"
              class="red arrow down icon"
            />
             (222333)
          </div>
          <div
            class="extra ExtraContent"
          >
            <span
              class="date"
            >
              01/01/1970 
            </span>
          </div>
        </div>
      </div>
      <div
        class="item padded-item RecentCall"
        data-testid="recent-1"
      >
        <i
          aria-hidden="true"
          class="grey phone large icon ui avatar"
        />
        <div
          class="content"
        >
          <div
            class="header ItemHeader"
          >
            123456
          </div>
          <div
            class="description Content"
          >
            <i
              aria-hidden="true"
              class="green arrow up icon"
            />
             (444555)
          </div>
          <div
            class="extra ExtraContent"
          >
            <span
              class="date"
            >
              01/01/1970 
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`RecentCallListContainer test renders empty list 1`] = `
<DocumentFragment>
  <div
    class="ScrollableContent"
  >
    <div
      class="ui link items"
      data-testid="RecentCallList"
    />
  </div>
</DocumentFragment>
`;
